### PR TITLE
Upgrade controller gen version to 0.12.0

### LINKED ISSUE #2150: Update controller-gen version
Would be good to update controller-gen to v0.8.0 (the latest). I think TrainedModel CRD generation had an issue before as discussed here: https://github.com/kubernetes-sigs/controller-tools/issues/391 when moving to newer versions of controller-gen, but I believe this was fixed in `knative/pkg` relatively recently through https://github.com/knative/pkg/pull/2431.

Can we update `knative/pkg` and subsequently controller-gen?

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2150 

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:
```release-note
Bump controller-gen version to 0.12.0
```

### CODE REVIEW

**Comment by sivanantha321** [comment]
Depends on #2914 for go version upgrade

File: Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
 
 ## Tool Versions
 KUSTOMIZE_VERSION ?= v5.0.3
-CONTROLLER_TOOLS_VERSION ?= v0.4.0
+CONTROLLER_TOOLS_VERSION ?= v0.12.0
```
Comment (yuzisun) [inline Makefile:42]: @sivanantha321 Do we need to regenerate the manifests with this version change?

Comment (sivanantha321) [reply Makefile:42]: yes Dan. I will regenerate the manifests

**Comment by yuzisun** [comment]
Thanks @sivanantha321 !

/lgtm
/approve

**Pull request merged**